### ISSUE TITLE
Add data based tests

### ISSUE THREAD

**Issue by tarioch** [johannesjh/smart_importer#18]
I think it would be great to have data based tests to cover lots of use cases.
I'm having some really bad predictions which I would like to narrow down with that.

@johannesjh would you have an issue if I switch over to pytest as this has some support for test generators

**Comment by johannesjh** [comment]
One consideration is which test frameworks are used in fava and beancount (to ease future integration with their code). Beancount's regression tests yield generated nose tests, see `beancount/beancount/ingest/regression.py`. So if nose suits your requirements, it would probably be the better choice.

**Comment by tarioch** [comment]
Makes sense. On my importers I use pytest to run the nose tests as it looks like nose is pretty much end of life. I might actually for now even go with the simplest case and not use any generator and simply add a method with the arguments for each test case

**Comment by johannesjh** [comment]
sounds good, it is sometimes best to keep it simple.
